presto-main/src/main/java/com/facebook/presto/operator/scalar/UrlFunctions.java

```diff
@@ -129,6 +132,42 @@ public static Slice urlExtractParameter(@SqlType(StandardTypes.VARCHAR) Slice ur
         return null;
     }
 
+    @Nullable
+    @Description("encode url")
+    @ScalarFunction
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice urlEncode(@SqlType(StandardTypes.VARCHAR) Slice url)
+    {
+        if (url == null) {
+            return null;
+        }
+
+        try {
+            return slice(URLEncoder.encode(url.toString(UTF_8), UTF_8.name()));
```

Comment:
Use Guava's `UrlEscapers.urlFormParameterEscaper`. It's probably a bit faster due to not having to look up the charset every time and doesn't throw UnsupportedEncodingException